app/controllers/course/video/topics_controller.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+class Course::Video::TopicsController < Course::Video::Controller
+  load_and_authorize_resource :topic, class: Course::Video::Topic.name
+  before_action :load_existing_topic, only: :create
+
+  include Course::Discussion::PostsConcern
+  skip_load_and_authorize_resource :post, except: :create
+
+  def index
+    @topics = @video.topics.includes(posts: :children).order(:timestamp)
+    @topics = @topics.reject { |topic| topic.posts.empty? }
+    @posts = @topics.map(&:posts).inject(Course::Discussion::Post.none, :+)
+  end
+
+  def create
+    result = @topic.class.transaction do
+      @post.title = @video.title
+      raise ActiveRecord::Rollback unless @post.save && create_topic_subscription && update_topic_pending_status
+      raise ActiveRecord::Rollback unless @topic.save
+      true
+    end
+
+    head :bad_request unless result
+  end
+
+  def show
+  end
+
+  private
+
+  def topic_params
```

Comment:
Actually is the `topic_params` and `topic_param` required? Seems like it is not used here.

Reply:
Or did you forget to call them in `timestamp_param` and `topic_param`?

Reply:
Nope, I think I forgot to remove them when I altered the API, will remove.